core/src/wallet/common/AbstractWallet.cpp

```diff
@@ -48,11 +48,11 @@ namespace ledger {
                                        const api::Currency &currency,
                                        const std::shared_ptr<WalletPool> &pool,
                                        const std::shared_ptr<DynamicObject> &configuration,
-                                       const DerivationScheme &derivationScheme
-        )
-                : DedicatedContext(
-                pool->getDispatcher()->getSerialExecutionContext(fmt::format("wallet_{}", walletName))),
-                  _scheme(derivationScheme) {
+                                       const DerivationScheme &derivationScheme)
+                : DedicatedContext(pool->getDispatcher()->getThreadPoolExecutionContext(fmt::format("wallet_{}", walletName))),
+                  _scheme(derivationScheme),
+                  _balanceCache(std::chrono::seconds(configuration->getInt(api::Configuration::TTL_BLOCK_CACHE).value_or(30)))
```

Comment:
I'm wandering if it's a good idea to put a fallback value to something which must be defined :
1. the default value `30` can be far away from the configuration value then introduce weird/unwanted/tricky behaviour
2. missing configuration value should be considered as technical bug and fix as soon as possible

Reply:
We should always provide default values when there is a rationale behind it, otherwise I agree we should not set any and make it a mandatory config, in that case if it's missing we throw an exception for that ! 
Here 30sec seems reasonable for both balance and blocks since it is inferior to the frequency of blocks mining